app.js

```diff
@@ -0,0 +1,281 @@
+    let productsList;
+    if (localStorage.getItem(productDetails) == null) {
+      productsList = [];
+    } else {
+      productsList = productData;
+    }
+    productsList.push({
+      productId: parseInt(userInput.id),
+      productName: userInput.name,
+      image: displayImage.src,
+      price: Number(userInput.price).toFixed(2),
+      description: userInput.description,
+    });
+    localStorage.setItem(productDetails, JSON.stringify(productsList));
+    window.location.href = "index.html";
+  }
+}
+
+function viewProduct() {
+  displayProduct(productData);
+}
+
+function deleteProduct(clickedId) {
+  let productsList = productData;
+  let msg = "Are you sure to remove this product?";
+  if (productsList != null) {
+    if (confirm(msg)) {
+      productsList.splice(clickedId, 1);
+      localStorage.setItem(productDetails, JSON.stringify(productsList));
+    }
+  }
+  viewProduct();
+}
+
+function updateProduct(index) {
+  btnText.innerHTML = "Update";
+  formTitle.innerHTML = "Update Product";
+  productId.disabled = true;
+  let productsList = productData;
+  if (productsList != null) {
+    productId.value = productsList[index].productId;
+    productName.value = productsList[index].productName;
+    displayImage.src = productsList[index].image;
+    productPrice.value = productsList[index].price;
+    productDescription.value = productsList[index].description;
+    objectIndex.value = index;
+  }
+  document.querySelector("#btn-operation").onclick = function () {
+    if (btnText.innerHTML == "Update") {
+      if (validateData() && nameValidation(index)) {
+        uploadDetails(index);
+      }
+    } else {
+      addProduct();
+    }
+  };
+}
+
+function uploadDetails(index) {
+  let setInput = new formData();
+  let productsList = productData;
+  productsList[index].productId = parseInt(setInput.id);
+  productsList[index].productName = setInput.name;
+  productsList[index].image = displayImage.src;
+  productsList[index].price = Number(setInput.price).toFixed(2);
+  productsList[index].description = setInput.description;
+  localStorage.setItem(productDetails, JSON.stringify(productsList));
+  $("#form-modal").hide();
+  $(".modal-backdrop").hide();
+  $(".toast").toast("show");
+  viewProduct();
+}
+
+function imageUrl(input) {
+  if (imageValidation()) {
+    if (input.files && input.files[0]) {
+      var reader = new FileReader();
+      reader.onload = function (e) {
+        displayImage.setAttribute("src", e.target.result);
+      };
+      reader.readAsDataURL(input.files[0]);
+    }
+  }
+}
+
+//validation functions
+function validateNameId() {
+  let userInput = new formData();
+  let productsList = productData || [];
+  let filterData = productsList.filter((element) => element.productId == userInput.id || element.productName == userInput.name);
+  let productLength = Object.keys(filterData).length;
+  if (productLength > 0) {
+    let msg = "Product ID and Name should be unique";

```

Comment:
This is wrong error message. ie. Id different but same product name then it is valid message?